addons/csw/functions/fnc_loadAmmo.sqf

```diff
@@ -0,0 +1,62 @@
+/*
+ * Author: TCVM
+ * Loads ammo into the CSW and removes it from the player inventory
+ *
+ * Arguments:
+ * 0: CSW <OBJECT>
+ * 1: Unit <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [CSW, player] call ace_csw_fnc_loadAmmo
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params["_csw", "_player"];
+
+private _cswWeapon = getArray(configFile >> "CfgVehicles" >> typeof(_csw) >> "Turrets" >> "MainTurret" >> "weapons") select 0;
+private _cswMagazines = getArray(configFile >> "CfgWeapons" >> _cswWeapon >> "magazines");
+
+private _magazineSelected = "";
+private _magazineAmmoCount = 0;
+private _maxAmmo = [_csw] call FUNC(getMaxAmmo);
+
+{
+    if ((_x select 0) in _cswMagazines && (_csw ammo _cswWeapon) + (_x select 1) <= _maxAmmo) exitWith {
```

Comment:
Please make use of lazy evaluation for all conditions. For example, this if statement would need to be changed to:

```
if ((_x select 0) in _cswMagazines && {(_csw ammo _cswWeapon) + (_x select 1) <= _maxAmmo}) exitWith {
```
You would add `{..}` for all tests that have multiple conditions. One for each condition after the first.

See also: https://community.bistudio.com/wiki/a_%26%26_b